overlord/snapstate/snapstate_test.go

```diff
@@ -4228,76 +4229,72 @@ run-hook: Hold`)
 func (s *snapmgrTestSuite) verifyRefreshLast(c *C) {
 	var lastRefresh time.Time
 
-	tr := config.NewTransaction(s.state)
-	tr.Get("core", "refresh.last", &lastRefresh)
+	s.state.Get("last-refresh", &lastRefresh)
 	c.Check(time.Now().Year(), Equals, lastRefresh.Year())
 }
 
+func makeTestRefreshConfig(st *state.State) {
+	now := time.Now()
+	st.Set("last-refresh", time.Date(2009, 8, 13, 8, 0, 5, 0, now.Location()))
+
+	tr := config.NewTransaction(st)
+	tr.Set("core", "refresh.schedule", fmt.Sprintf("00:00-23:59"))
+	tr.Commit()
+}
+
 func (s *snapmgrTestSuite) TestEnsureRefreshesNoUpdate(c *C) {
 	s.state.Lock()
 	defer s.state.Unlock()
 	snapstate.CanAutoRefresh = func(*state.State) (bool, error) { return true, nil }
 
-	tr := config.NewTransaction(s.state)
-	tr.Set("core", "refresh.last", time.Time{})
-	tr.Commit()
+	makeTestRefreshConfig(s.state)
 
 	// Ensure() also runs ensureRefreshes()
 	s.state.Unlock()
 	s.snapmgr.Ensure()
+	// give the timer a bit of time
+	time.Sleep(100 * time.Millisecond)
```

Comment:
This will likely create issues in slow systems. If last-refresh is supposed to get updated, can we test this inside a time-bounded loop with shorter sleeps to decide when we're done waiting?